repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java

```diff
@@ -2865,61 +2865,54 @@ public List<String> propagateClassification(String entityGuid, String classifica
                 return null;
             }
 
-            List<String> propagatedEntitiesGuid = new ArrayList<>();
-            long impactedVerticesSize = impactedVertices.size();
-            int offset = 0;
-
-            LOG.info(String.format("Total number of vertices to propagate: %d", impactedVerticesSize));
-            AtlasPerfMetrics.MetricRecorder chunkedPropMetricRecorder = RequestContext.get().startMetricRecord("chunkedPropagationAndNotification");
-
-            int toIndex;
-
-            do {
-                toIndex = ((offset + CHUNK_SIZE > impactedVerticesSize) ? (int) impactedVerticesSize : (offset + CHUNK_SIZE));
-
-                List<String> chunkedGuids = processChunkedPropagation(impactedVertices.subList(offset, toIndex), classificationVertex);
-
-                transactionInterceptHelper.intercept();
-
-                if((chunkedGuids != null) && (! chunkedGuids.isEmpty())){
-                    propagatedEntitiesGuid.addAll(chunkedGuids);
-                }
-                offset += CHUNK_SIZE;
-            }
-            while (offset < impactedVerticesSize);
-
-            RequestContext.get().endMetricRecord(chunkedPropMetricRecorder);
-
-            return propagatedEntitiesGuid;
+            return processClassificationPropagationAddition(impactedVertices, classificationVertex);
         } catch (Exception e) {
             LOG.error("propagateClassification(entityGuid={}, classificationVertexId={}): error while propagating classification", entityGuid, classificationVertexId, e);
 
             throw new AtlasBaseException(e);
         }
     }
 
-    public List<String> processChunkedPropagation(List<AtlasVertex> chunkedVerticesToPropagate, AtlasVertex classificationVertex) throws AtlasBaseException{
-        try {
-            AtlasPerfMetrics.MetricRecorder metricRecorder = RequestContext.get().startMetricRecord("lockObjectsAfterTraverse");
-            List<String> impactedVerticesGuidsToLock = chunkedVerticesToPropagate.stream().map(x -> GraphHelper.getGuid(x)).collect(Collectors.toList());
-            RequestContext.get().endMetricRecord(metricRecorder);
+    public List<String> processClassificationPropagationAddition(List<AtlasVertex> verticesToPropagate, AtlasVertex classificationVertex) throws AtlasBaseException {
+        AtlasPerfMetrics.MetricRecorder chunkedPropMetricRecorder = RequestContext.get().startMetricRecord("chunkedPropagationAndNotification");
```

Comment:
Change metric name to processClassificationPropagationAddition

Reply:
Changed it